src/mono/mono/mini/mini-llvm.c

```diff
@@ -10501,6 +10517,24 @@ add_intrinsic (LLVMModuleRef module, int id)
 	case INTRINS_BITREVERSE_I64:	
 		intrins = add_intrins1 (module, id, LLVMInt64Type ());	
 		break;	
+	case INTRINS_ADV_SIMD_ABS_FLOAT:
+		intrins = add_intrins1 (module, id, sse_r4_t);
+		break;
+	case INTRINS_ADV_SIMD_ABS_DOUBLE:
+		intrins = add_intrins1 (module, id, sse_r8_t);
+		break;
+	case INTRINS_ADV_SIMD_ABS_INT8:
+		intrins = add_intrins1 (module, id, sse_i1_t);
+		break;
+	case INTRINS_ADV_SIMD_ABS_INT16:
+		intrins = add_intrins1 (module, id, sse_i2_t);
+		break;
+	case INTRINS_ADV_SIMD_ABS_INT32:
+		intrins = add_intrins1 (module, id, sse_i4_t);
+		break;
+	case INTRINS_ADV_SIMD_ABS_INT64:
+		intrins = add_intrins1 (module, id, sse_i8_t);
+		break;
```

Comment:
The whole function looks like the work to define signatures for intrinsics could be done via macros in `llvm-intrinsics.h` instead.

Reply:
Do we have macros to do this type of thing, or were you suggesting I create some? I would have think think a little about how to do it; this function predates my changes and would need some reworking.

Reply:
Yeah, I mean in general, if you want to eventually implement the full set of intrinsics it makes sense to minimize verbosity of the changes.